runner.go

```diff
@@ -91,20 +110,24 @@ func Run(tasks []*Task, options ...RunOption) {
 	ctx, cancel := context.WithCancel(context.Background())
 	onInterruptSignal(cancel)
 
-	g, ctx := errgroup.WithContext(ctx)
+	for _, hook := range runOptions.onStartHooks {
+		hook(ctx, executor)
+	}
 
-	// Reporters should use a different context because we want to stage
-	// their cancellation after the executor itself has been completed.
-	reporterCtx, cancelReporter := context.WithCancel(context.Background())
-	for _, reporterFn := range runOptions.ReporterFns {
+	g, ctx := errgroup.WithContext(ctx)
+	for i := range runOptions.subscriptions {
+		sub := runOptions.subscriptions[i]
```

Comment:
This bug was fun 👍

Reply:
Maybe I should add a comment in the code:

The variable at the top (`i`/`reporterFn`) is re-assigned for each entry. In order for the goroutine to use the right function we have to allocate a variable in the for scope.